src/main/java/com/czertainly/core/service/cmp/util/CertUtil.java

```diff
@@ -0,0 +1,151 @@
+package com.czertainly.core.service.cmp.util;
+
+import org.bouncycastle.asn1.ASN1Encoding;
+import org.bouncycastle.asn1.ASN1OctetString;
+import org.bouncycastle.asn1.DEROctetString;
+import org.bouncycastle.asn1.cmp.CMPCertificate;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.security.*;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.czertainly.core.service.cmp.message.PkiMessageDumper.ifNotNull;
+
+public class CertUtil {

```

Comment:
Duplicate, use utils in package `com.czertainly.core.util`.

Reply:
DONE, removed (part of implementation move into existing CertificateUtil.java)